src/modsim2/similarity/metrics/pad.py

```diff
@@ -0,0 +1,322 @@
+        # Split A&B into train and test
+        train_A, test_A = train_test_split(data_A, test_size=test_size, shuffle=True)
+        train_B, test_B = train_test_split(data_B, test_size=test_size, shuffle=True)
+
+        return train_A, test_A, train_B, test_B
+
+    @staticmethod
+    def __label_data(data: np.ndarray, label_value: int) -> np.ndarray:
+        """
+        Takes an array of data and an integer label value. Returns
+        an array of the label value with length matching the first
+        dimension of the array of data
+
+        Args:
+            data: array of data (any dimension)
+            label_value: integer value to set the label data
+
+        Returns:
+            labels: array of value label_value with length matching
+                    the number of data records
+        """
+        labels = np.ones(data.shape[0])
+        labels *= label_value
+        return labels
+
+    def __concat_data(
+        self,
+        train_A: np.ndarray,
+        test_A: np.ndarray,
+        train_B: np.ndarray,
+        test_B: np.ndarray,
+    ):
+        """
+        This method will label the A and B train and tests datasets,
+        then concatenate the train and test data. The train data are
+        then shuffled.
+        The data from A will be labelled 1 and B will be labelled 0
+        The train and test data and labels are assigned to class attributes
+
+        Args:
+            train_A: the data for training from dataset A
+            test_A: the data for testing from dataset A
+            train_B: the data for training from dataset B
+            test_B: the data for testing from dataset B
+        """
+        # Create labels for A & B data
+        train_labels_A, test_labels_A = self.__label_data(
+            train_A, 1
+        ), self.__label_data(test_A, 1)
+        train_labels_B, test_labels_B = self.__label_data(
+            train_B, 0
+        ), self.__label_data(test_B, 0)
+
+        # Concatenate the A and B datasets into train and test
+        self.__train_data = np.concatenate((train_A, train_B), axis=0)
+        self.__train_labels = np.concatenate((train_labels_A, train_labels_B), axis=0)
+        self.__test_data = np.concatenate((test_A, test_B), axis=0)
+        self.__test_labels = np.concatenate((test_labels_A, test_labels_B), axis=0)
+        # Shuffle the train data
+        self.__train_data, self.__train_labels = shuffle(
+            self.__train_data, self.__train_labels
+        )
+
+    def __transform_data(self):
+        """
+        Method to transform the object's train and test data
+        using the object's embedding function
+        The transformed data are assigned to attributes of the object
+        """
+        # Extract embedding callable
+        embedding_fn = EMBEDDING_FN_DICT[self.__embedding_name]
+        # Transform the train and test data using the embedding function
+        self.__embed_train_data = embedding_fn(self.__train_data)
```

Comment:
quite a lot of wrapper for a side effect here. I think it's probably clearer just to do it inline as only called once (and rm this fn)

Reply:
Sure, I have removed this.